src/usb_hid.py

```
@@ -894,18 +961,26 @@ def enable(requested_devices: Sequence[Device], boot_device: int = 0) -> None:
+            )
             try:
                 Path("%s/" % function_root).mkdir(parents=True)
             except FileExistsError:
                 continue
+            device._report_id_to_function_instance[report_id] = function_instance
+            function_instance_index += 1
             Path("%s/protocol" % function_root).write_text(
-                "%s" % report_id, encoding="utf-8"
+                "%s" % device.protocol, encoding="utf-8"
             )
             Path("%s/report_length" % function_root).write_text(
```

Ruff (0.15.11)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)